### PR TITLE
Fix handling of resource definitions in stack apply

### LINKED ISSUE #13: Stack apply fails for stacks containing both CRD + instances of that custom resource
The `K8s::Stack#apply` => `client.get_resources` attempts to perform API discovery for all stack resources before starting the POST the resources. This fails for custom resources associated with CRDs in the same stack because the CRDs are not defined yet, so the API discovery returns a HTTP 404.

```
GET /apis/certmanager.k8s.io/v1alpha1 => HTTP 404 Not Found: 404 page not found
/usr/local/bundle/gems/k8s-client-0.3.0/lib/k8s/transport.rb:171:in `parse_response'
/usr/local/bundle/gems/k8s-client-0.3.0/lib/k8s/transport.rb:215:in `block in requests'
/usr/local/bundle/gems/k8s-client-0.3.0/lib/k8s/transport.rb:213:in `map'
/usr/local/bundle/gems/k8s-client-0.3.0/lib/k8s/transport.rb:213:in `requests'
/usr/local/bundle/gems/k8s-client-0.3.0/lib/k8s/transport.rb:254:in `gets'
/usr/local/bundle/gems/k8s-client-0.3.0/lib/k8s/client.rb:72:in `apis'
/usr/local/bundle/gems/k8s-client-0.3.0/lib/k8s/client.rb:123:in `get_resources'
/usr/local/bundle/gems/k8s-client-0.3.0/lib/k8s/stack.rb:72:in `apply'
/app/lib/pharos/addon.rb:205:in `apply_resources'
/app/lib/pharos/addon.rb:171:in `apply_install'
/app/lib/pharos/addon.rb:157:in `apply'
/app/lib/pharos/cluster_manager.rb:126:in `block in apply_addons'
/app/lib/pharos/addon_manager.rb:74:in `block in each'
/app/lib/pharos/addon_manager.rb:86:in `block in with_enabled_addons'
/app/lib/pharos/addon_manager.rb:83:in `each'
/app/lib/pharos/addon_manager.rb:83:in `with_enabled_addons'
/app/lib/pharos/addon_manager.rb:70:in `each'
/app/lib/pharos/cluster_manager.rb:123:in `apply_addons'
/app/lib/pharos/up_command.rb:107:in `configure'
/app/lib/pharos/up_command.rb:44:in `block in execute'
/app/lib/pharos/up_command.rb:43:in `chdir'
/app/lib/pharos/up_command.rb:43:in `execute'
/usr/local/bundle/gems/clamp-1.2.1/lib/clamp/command.rb:63:in `run'
/usr/local/bundle/gems/clamp-1.2.1/lib/clamp/subcommand/execution.rb:11:in `execute'
/usr/local/bundle/gems/clamp-1.2.1/lib/clamp/command.rb:63:in `run'
/usr/local/bundle/gems/clamp-1.2.1/lib/clamp/command.rb:132:in `run'
/app/lib/pharos/root_command.rb:16:in `run'
bin/pharos-cluster:12:in `<main>'
```

The API discovery at this point should just ignore the 404 errors, and assume that those resources will not exist yet.

### PR DESCRIPTION
**Pull request by jakolehm**
Fixes #13

A bit more simpler alternative for #14.

### CODE REVIEW

File: lib/k8s/stack.rb [hunk 1]
```diff
@@ -67,8 +71,21 @@ def prepare_resource(resource, base_resource: nil)
       })
     end
 
+    # @param client [K8s::Client] apply using client
+    # @param prune [Boolean] prune automatically after apply
     # @return [Array<K8s::Resource>]
     def apply(client, prune: true)
+      specials, others = resources.partition { |r| SPECIAL_KINDS.include?(r.kind) }
```
Comment (SpComb) [inline lib/k8s/stack.rb]: "Resource Definitions" would be a better term than "special" - these are resources that **define** other resource kinds.

Comment (jakolehm) [reply lib/k8s/stack.rb]: Fixed.

File: lib/k8s/stack.rb [hunk 2]
```diff
@@ -7,6 +7,10 @@ class Stack
 
     LABEL = 'k8s.kontena.io/stack'
     CHECKSUM_ANNOTATION = 'k8s.kontena.io/stack-checksum'
+    RESOURCE_DEFINITIONS = [
+      'CustomResourceDefinition',
```
Comment (SpComb) [inline lib/k8s/stack.rb]: I don't think `kind` is necessarily unique across the entire API, so strictly speaking these whitelisted resource types should be qualified by the `apiGroup`, as for `PRUNE_IGNORE`...  so these should look something like:

* `apiextensions.k8s.io:CustomResourceDefinition` 
* `apiregistration.k8s.io:APIService`

However, including the complete `apiVersion` (`apiregistration.k8s.io/v1beta1`) would be a bit hazardous, just the apigroup itself should be enough.

**Comment by jakolehm** [comment]
#14 starts to look simple enough.

**Pull request closed without merging**